### DISCUSSION TITLE
some of the codespaces I created cannot display code highlight on my custom language

### DISCUSSION THREAD

**Discussion by amyXia1994** [community/community#42768] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

I created codespaces now and then, and some of them can not display the code highlight on my custom KCL language.

I've added the KCL language extension to my devcontainer file. But sometimes the extension seems not has been activated. I tried de-activate and re-activate it in the `Extensions` view, doesn't work.

There's one of the codespace which does not display code highlight accurately: https://amyxia1994-studious-lamp-qvj4gjgv6g7c9rx6.github.dev/

You can create the same codespace [here](https://github.com/KusionStack/konfig) by clicking the [Open in GitHub Codespaces] badge in readme file. **I've tried for two days now and then, NONE of the codespaces created display code highlight correctly**. I'm currently recording a demo video to teach how to use my product and the vscode extension on the GitHub Codespaces, but the code highlight keeps being not showing and my demo work just stucked here. 😞 

Here's how it looks:
![image](https://user-images.githubusercontent.com/29722609/209789952-fed13048-53f7-448e-b27c-1e742d2d5a6b.png)

Here's how it looks normally:

![view-code](https://user-images.githubusercontent.com/29722609/209790977-40f7a431-6929-472e-bdd2-f075ee494a94.gif)

**Comment by cmbrose** [comment, marked as answer]
👋 Thanks for reaching out @amyXia1994 - I just created a codespace for your repo and syntax highlighting worked for me and I can see your extension was installed - does this issue still repro for you? Since you own the syntax highlighting extension, can you check if there are any issues in activation that may cause the syntax highlighting to fail?

![image](https://user-images.githubusercontent.com/5447118/210090157-71ad1c55-a606-4b9f-88f0-313c64bd474d.png)

**Reply by amyXia1994** [reply]
@cmbrose Hi cmbrose, Thanks for your prompt reply! 

It seems that the issue only reproduced on the two days before I reported it(Before I reported that, I tried during two days creating several codespaces and the issue constantly reproduces. And I'm sorry that I deleted that problematic codespace to save my storage quota so I can't refresh and check it again). But After that day, I re-tried creating the codespace several times, and for now, that issue did not produce again. 😌 

I used to install the extension to my local VS Code and that activation problem hasn't happened that way, so I'm not sure if that's related to some issues with my extension.  Do you have some suggestions about what I might check on? and in case the issue happens again, does saving(by that I mean not deleting) the codespace with the activating issue helps to locate the cause?

**Reply by cmbrose** [reply]
If it does repro again, the most helpful info would be the logs from your extension and the vscode extension manager I think. Leaving the codespace around may help us try some recovery steps, but codespaces devs don't have access to use your codespace ourselves for security/privacy reasons. It may have been a transient issue with vscode extensions and may not repro again, but please let us know if it does.

**Reply by amyXia1994** [reply]
I'll check the extension log and report the issue if that reproduces. Thanks for your help!
